### PR TITLE
v0.1.0

### PR DESCRIPTION
**Pull request by levb**
v0.1.0: updated for some known issues, including #2 

Enhancements:
- Respect Root and Parent post IDs - `/giphy` works in threaded conversations now 
- "Shuffle" now tries 3 times to get a different URL
- Display the giphy page URL in the message
- Updated README.md

Bugs and technical improvements:
- Moved RegisterCommand to OnActivate to avoid deadlocks in
  OnConfigurationChange
- Updated Makefile with the experimental debug changes from the Jira
  plugin
- Removed the `Trigger` config option as unneccessary
- Removed server version check from the Go code, added
  min_server_version: 5.12.0 to the manifest.
- Restructured the go code to be more intention-revealing

### CODE REVIEW

File: server/giphy.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+func Query(cf Config, s string) (linkURL, embedURL string, err error) {
+	q := url.Values{}
+	q.Set("api_key", cf.APIKey)
+	q.Set("s", s)
+	q.Set("weirdness", "10")
```
Comment (mickmister) [inline server/giphy.go:14]: Is there a reason we hardcode the `weirdness` to 10?

Comment (levb) [reply server/giphy.go:14]: LOL, no, filed https://github.com/levb/mattermost-plugin-giphy/issues/8.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,66 @@
...
+ - [4. FAQ](#4-frequently-asked-questions-faq).
+ - [5. Help](#5-help).
+ 
+### Requirements
+- For Giphy Plugin 1.0, Mattermost Server v5.14+ is required
+- Giphy plugin requires configuring a [Giphy API Key](https://developers.giphy.com/faq). 
+
+## 1. Users
+
+The giphy plugin implements a `/giphy` command which can be used to look and post images for phrases.
```
Comment (mickmister) [inline README.md]: Nit: `s/look/view`

**Review by mickmister — APPROVED** [review]

**Review by cpoile — APPROVED** [review]

**Comment by cpoile** [comment]
@levb there's a build error?

**Comment by levb** [comment]
@cpoile CI is not set up yet, I'd like to transfer the repo to mattermost first.

**Comment by levb** [comment]
Also, this resolves https://github.com/levb/mattermost-plugin-giphy/issues/2

**Pull request merged**